File: 백준/Gold/13144. List of Unique Numbers/List of Unique Numbers.java
import java.io.*;
import java.util.*;


public class Main {



    public static void main(String[] args) throws NumberFormatException, IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        
        int [] arr = new int[100001];
        int [] visited = new int [100001];

        StringTokenizer st = new StringTokenizer(br.readLine());

        int arrindex=0;

        while(st.hasMoreTokens()){
            arr[arrindex]= Integer.parseInt(st.nextToken());
            arrindex++;
        }

        int end = 0;
        long res = 0;

        for (int i = 0; i < n ; i++) {
            while(end<n){
                if(visited[arr[end]]==1)break;
                visited[arr[end]]=1;
                end++;
            }
            res+=(end-i);
            visited[arr[i]]=0;
        }

        System.out.println(res);



    }
}

File: 백준/Gold/13144. List of Unique Numbers/List of Unique Numbers.cc
#include<bits/stdc++.h>
using namespace std;


int n;
long long int res;
int visited[100001];


int main() {

    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    cin >> n;
    vector<int>arr(n);

    for(int i =0; i < n ;i++){cin >> arr[i];}

    int end = 0;
    for(int start = 0 ; start < n ; start++){
        while(end < n){
            if(visited[arr[end]])break;
            visited[arr[end]]=1;
            end++;
        }
        res += (end - start);
        visited[arr[start]]=0;
    }
    cout << res <<"\n";

    return 0;

}